internal/provider/common.go

```diff
@@ -26,33 +27,65 @@ type NodeClientCollection interface {
 	getNodeClient(sub *substrate.Substrate, nodeID uint32) (*client.NodeClient, error)
 }
 
+type Progress struct {
+	time    time.Time
+	stateOk int
+}
+
+func getExponentialBackoff(initial_interval time.Duration, multiplier float64, max_interval time.Duration, max_elapsed_time time.Duration) *backoff.ExponentialBackOff {
+	b := backoff.NewExponentialBackOff()
+	b.InitialInterval = initial_interval
+	b.Multiplier = multiplier
+	b.MaxInterval = max_interval
+	b.MaxElapsedTime = max_elapsed_time
+	return b
+}
+
 func waitDeployment(ctx context.Context, nodeClient *client.NodeClient, deploymentID uint64, version uint32) error {
-	done := false
-	for start := time.Now(); time.Since(start) < 4*time.Minute; time.Sleep(1 * time.Second) {
-		done = true
+	last_progress := Progress{time.Now(), 0}
+	var deploy_error error = nil
+	backoff.Retry(func() error {
+		done := true
+		state_Ok := 0
+
 		sub, cancel := context.WithTimeout(ctx, 10*time.Second)
 		defer cancel()
 		dl, err := nodeClient.DeploymentGet(sub, deploymentID)
 		if err != nil {
-			return err
+			deploy_error = err
+			return nil
 		}
 		if dl.Version != version {
-			continue
+			return errors.New("not the same version")
```

Comment:
Since this will appear to the user "the deployment version wasn't updated on the node" would be more descriptive?